root/scripts/utils.js

```diff
@@ -281,38 +281,64 @@ export function recipeIdArrayToObject(arr) {
  * @returns An array of recipeObjects that matches the search parameters
  */
 export async function search(searchQuery, tags) {
-  // match query to title, ingredients
-  const searchResults = new Set();
+  // Ignore recipe if query matches less than this percent of title
+  const MIN_MATCHING_THRESHOLD = 0.5;
   const allRecipes = await getAllRecipes();
   const query = searchQuery.toLowerCase();
+  const tokenizedQuery = [...new Set(query.trim().split(/\s+/))]; // regex matches one or more spaces
+  const minNumMatchingTokens = Math.ceil(tokenizedQuery.length * MIN_MATCHING_THRESHOLD);
+
+  let searchResults = [];
 
   for (let i = 0; i < allRecipes.length; i += 1) {
     const recipe = allRecipes[i];
     let recipeMatches = true;
     try {
+      // Check that recipe matches tags, reject if it doesn't
       tags.forEach((tag) => {
         if (!recipe[`${tag}`]) {
           recipeMatches = false;
         }
       });
 
+      // Create a search score for the title
+      let numMatchingTokens = 0;
+      let mostRecentMatch = tokenizedQuery.length;
+      let numMatchingCharacters = 0;
+      let searchScore = 0;
       const { title } = recipe;
       if (title) {
-        if (!title.toLowerCase().includes(query)) {
-          recipeMatches = false;
+        for (let j = tokenizedQuery.length - 1; j >= 0; j -= 1) {
+          if (title.toLowerCase().includes(tokenizedQuery[j])) {
```

Comment:
This is a LOT more robust than what we had before. I was doing some testing and looks like "burger with garlic aioli" and "vegan pudding" work as search queries with this new algorithm. Where as, the old one did not display anything. I take it that matching each token (individual word) against the title is what allowed this.

For an even more robust algorithm, what if we also factored the `ingredients-list` as a factor in our search scores (like what we're doing with the title here). Let me know your thoughts, we can obviously open up a new ticket for this if time is short.

Reply:
I think it'd be best to open a new ticket for ingredient search